konig-schemagen/src/main/java/io/konig/schemagen/sql/RdbmsShapeGenerator.java

```diff
@@ -172,9 +166,10 @@ private void process(Shape rdbmsShape, String logicalPath, String prefix, Shape
 	}
 	
 	private void addSyntheticKey(Shape rdbmsShape, String suffix, URI relationshipProperty) throws RDFParseException, IOException {
-		PropertyConstraint pc = hasPrimaryKey(rdbmsShape);
+		Shape shape = rdbmsShape.getRdbmsLogicalShape();
+		PropertyConstraint pc = hasPrimaryKey(shape);
 		String localName = "";
-		if(pc != null) {
+		if(pc != null && "_PK".equals(suffix)) {
```

Comment:
Why are we testing that the suffix must equal "_PK"?

Reply:
This is to avoid primary keys getting "_FK" as suffix in the generated RDBMS shape.